src/main/java/updater/UpdateManager.java

```diff
@@ -90,6 +103,17 @@ private boolean initUpdate() {
             return false;
         }
 
+        File replaceAndExecuteJar = new File(UPDATE_DIRECTORY + File.separator + UPDATE_RESTARTER_APP_NAME);
+        if (!replaceAndExecuteJar.exists()) {
```

Comment:
Could we invert this? So if it exists, we return true, then outside the block we can do all the rest.